gps/app/views/Settings.js

```diff
@@ -55,11 +54,45 @@ export const SettingsScreen = ({ navigation }) => {
     };
   }, [navigation]);
 
+  //  This method is used to track whether user has "Allow" or "Deny" location permission system Dialog.
+  const systemDialogPermission = async () => {
+    BackgroundGeolocation.on('authorization', async status => {
+      if (status === BackgroundGeolocation.AUTHORIZED) {
+        // when user allow
+        setIsLogging(true);
+        LocationServices.start();
+        await SetStoreData(PARTICIPATE, true);
+      } else {
+        // when user deny
+        setIsLogging(false);
+        LocationServices.stop();
+        await SetStoreData(PARTICIPATE, false);
+      }
+    });
+  };
+
+  // checking whether location permission is enable or disbale from application setting on load of setting's page.
+  const checkIsLocationEnable = () => {
+    LocationService.checkStatus()
+      .then(async res => {
+        if (res !== undefined) {
```

Comment:
this line can be removed and the next line can just be 
```js
if (res.isRunning) {
```

Reply:
Hi, TCL is a Turbo console log extension, use as a shortcut to create console prints.